src/com/palmergames/bukkit/towny/command/TownyAdminCommand.java

```diff
@@ -1444,24 +1471,61 @@ public String nextNpcName() throws TownyException {
 				throw new TownyException(TownySettings.getLangString("msg_err_too_many_npc"));
 		} while (true);
 	}
+	
+	public void reloadLangs() {
+		String rootFolder = TownyUniverse.getInstance().getRootFolder();
+		try {
+			TownySettings.loadLanguage(rootFolder + File.separator + "settings", "english.yml");
+		} catch (IOException e) {
+			TownyMessaging.sendErrorMsg(sender, TownySettings.getLangString("msg_reload_error"));
+			e.printStackTrace();
+		}
+		
+		TownyMessaging.sendMsg(sender, TownySettings.getLangString("msg_reloaded_lang"));
+	}
+	
+	public void reloadPerms() {
+		String rootFolder = TownyUniverse.getInstance().getRootFolder();
+		TownyPerms.loadPerms(rootFolder + File.separator + "settings", "townyperms.yml");
+		TownyMessaging.sendMsg(sender, TownySettings.getLangString("msg_reloaded_perms"));
+	}
 
-	public void reloadTowny(Boolean reset) {
+	/**
+	 * Reloads only the config
+	 */
+	public void reloadConfig() {
+		try {
+			String rootFolder = TownyUniverse.getInstance().getRootFolder();
+			TownySettings.loadConfig(rootFolder + File.separator + "settings" + File.separator + "config.yml", plugin.getVersion());
+		} catch (IOException e) {
+			TownyMessaging.sendErrorMsg(sender, TownySettings.getLangString("msg_reload_error"));
+			e.printStackTrace();
+			return;
+		}
+		
+		TownyMessaging.sendMsg(sender, TownySettings.getLangString("msg_reloaded_config"));
+	}
 
+	/**
+	 * Reloads both the database and the config. Used with a database reload command.
+	 * 
+	 * @param reset Whether or not to reset the database.
+	 */
+	public void reloadDatabase(Boolean reset) {
 		if (reset) {
 			TownyUniverse.getInstance().getDataSource().deleteFile(plugin.getConfigPath());
 		}
 		if (plugin.load()) {
-			
+
 			// Register all child permissions for ranks
 			TownyPerms.registerPermissionNodes();
-			
+
 			// Update permissions for all online players
 			TownyPerms.updateOnlinePerms();
-			
+
 		}
 
-		TownyMessaging.sendMsg(sender, TownySettings.getLangString("msg_reloaded"));
-		// TownyMessaging.sendMsg(TownySettings.getLangString("msg_reloaded"));
+		TownyMessaging.sendMsg(sender, TownySettings.getLangString("msg_reloaded_db"));

```

Comment:
Should have a second language string saying the config.yml was reset to default.